projects/lib/src/interceptors/default-oauth.interceptor.ts

```diff
@@ -23,7 +24,14 @@ export class DefaultOAuthInterceptor implements HttpInterceptor {
     ) { }
 
     private checkUrl(url: string): boolean {
-        const found = this.moduleConfig.resourceServer.allowedUrls.find(u => url.startsWith(u));
+        if (!this.moduleConfig.resourceServer.allowedUrls && !this.moduleConfig.resourceServer.customUrlValidation) {
+            return true;
+        }
+
+        const found = !this.moduleConfig.resourceServer.customUrlValidation
+            ? this.moduleConfig.resourceServer.allowedUrls.find(u => url.startsWith(u))
+            : this.moduleConfig.resourceServer.customUrlValidation(url);
+
```

Comment:
I think you could simplify this a bit to avoid having to check the `customUrlValidation` multiple times:

```
if (this.moduleConfig.resourceServer.customUrlValidation) {
    return this.moduleConfig.resourceServer.customUrlValidation(url);
}

if (this.moduleConfig.resourceServer.allowedUrls) {
    return !!this.moduleConfig.resourceServer.allowedUrls.find(u => url.startsWith(u));
}

return true;
```

Reply:
I am not checking it multiple times. I am using "ternary conditional operator" and you are doing the same just in more longer way.

Reply:
agreed